loaders/embededKeyLoader_test.go

```diff
@@ -0,0 +1,314 @@
+	if m.err != nil {
+		return nil, m.err
+	}
+	if key, ok := m.keys[id]; ok {
+		return key, nil
+	}
+	return nil, errors.New("key not found")
+}
+
+func TestNewEmbeddedKeyLoader(t *testing.T) {
+	t.Run("default configuration", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		assert.True(t, loader.useCache)
+		assert.NotNil(t, loader.cache)
+		assert.Nil(t, loader.keyLoader)
+	})
+
+	t.Run("with custom loader", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{}
+		loader := NewEmbeddedKeyLoader(WithKeyLoader(mockLoader))
+		assert.True(t, loader.useCache)
+		assert.NotNil(t, loader.cache)
+		assert.Equal(t, mockLoader, loader.keyLoader)
+	})
+
+	t.Run("without cache", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader(WithCacheDisabled())
+		assert.False(t, loader.useCache)
+		assert.Nil(t, loader.cache)
+	})
+
+	t.Run("multiple options", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{}
+		loader := NewEmbeddedKeyLoader(
+			WithKeyLoader(mockLoader),
+			WithCacheDisabled(),
+		)
+		assert.False(t, loader.useCache)
+		assert.Nil(t, loader.cache)
+		assert.Equal(t, mockLoader, loader.keyLoader)
+	})
+}
+
+func TestEmbeddedKeyLoader_Load(t *testing.T) {
+	testKey := []byte("test-key-data")
+	testID := circuits.CircuitID("test-circuit")
+
+	t.Run("load from cache", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+		loader.storeInCache(testID, testKey)
+
+		key, err := loader.Load(testID)
+		require.NoError(t, err)
+		assert.Equal(t, testKey, key)
+	})
+
+	t.Run("load from custom loader", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{
+			keys: map[circuits.CircuitID][]byte{
+				testID: testKey,
+			},
+		}
+		loader := NewEmbeddedKeyLoader(WithKeyLoader(mockLoader))
+
+		key, err := loader.Load(testID)
+		require.NoError(t, err)
+		assert.Equal(t, testKey, key)
+
+		// Verify key was cached
+		cachedKey := loader.getFromCache(testID)
+		assert.Equal(t, testKey, cachedKey)
+	})
+
+	t.Run("custom loader error fallback to embedded", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{
+			err: errors.New("mock error"),
+		}
+		loader := NewEmbeddedKeyLoader(WithKeyLoader(mockLoader))
+
+		_, err := loader.Load(circuits.AuthV2CircuitID)
+		require.Error(t, err)
+	})
+
+	t.Run("no cache", func(t *testing.T) {
+		mockLoader := &MockKeyLoader{
+			keys: map[circuits.CircuitID][]byte{
+				testID: testKey,
+			},
+		}
+		loader := NewEmbeddedKeyLoader(
+			WithKeyLoader(mockLoader),
+			WithCacheDisabled(),
+		)
+
+		key, err := loader.Load(testID)
+		require.NoError(t, err)
+		assert.Equal(t, testKey, key)
+
+		// Verify key was not cached
+		assert.Nil(t, loader.cache)
+	})
+
+	t.Run("embedded key not found", func(t *testing.T) {
+		loader := NewEmbeddedKeyLoader()
+
+		_, err := loader.Load("non-existent-circuit")
+		assert.Error(t, err)
+		assert.Contains(t, err.Error(), "failed to load default key")
+	})
+}
+
+func TestDefaultEmbeddedKeys(t *testing.T) {
```

Comment:
golang table driven tests
https://dev.to/boncheff/table-driven-unit-tests-in-go-407b